BUILD.bazel

```diff
@@ -980,25 +937,215 @@ ray_cc_library(
         ],
         "//conditions:default": [],
     }),
+    deps = [
+        ":affinity_with_bundle_scheduling_policy",
+        ":bundle_scheduling_policy",
+        ":cluster_resource_manager",
+        ":cluster_resource_scheduler",
+        ":cluster_task_manager",
+        ":composite_scheduling_policy",
+        ":hybrid_scheduling_policy",
+        ":local_resource_manager",
+        ":node_affinity_scheduling_policy",
+        ":node_label_scheduling_policy",
+        ":random_scheduling_policy",
+        ":spread_scheduling_policy",
+    ],
+)
+
+ray_cc_library(
+    name = "scheduling_options",
+    hdrs = ["src/ray/raylet/scheduling/policy/scheduling_options.h"],
+    deps = [":ray_common"],
+)
+
+ray_cc_library(
+    name = "scheduling_context",
+    hdrs = ["src/ray/raylet/scheduling/policy/scheduling_context.h"],
+    deps = [
+        ":ray_common",
+        "@com_google_absl//absl/container:flat_hash_map",
+    ],
+)
+
+ray_cc_library(
+    name = "scheduler_internal",
+    hdrs = ["src/ray/raylet/scheduling/internal.h"],
+    deps = [
+        ":ray_common",
+        "//src/ray/protobuf:node_manager_cc_proto",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_resource_manager",
+    srcs = ["src/ray/raylet/scheduling/cluster_resource_manager.cc"],
+    hdrs = ["src/ray/raylet/scheduling/cluster_resource_manager.h"],
+    deps = [
+        ":local_resource_manager",
+        ":ray_common",
+        "@com_google_absl//absl/container:flat_hash_map",
+        "@com_google_googletest//:gtest",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_resource_scheduler",
+    srcs = [
+        "src/ray/raylet/scheduling/cluster_resource_scheduler.cc",
+    ],
+    hdrs = [
+        "src/ray/raylet/scheduling/cluster_resource_scheduler.h",
+    ],
+    deps = [
+        ":cluster_resource_manager",
+        ":composite_scheduling_policy",
+        ":scheduler_internal",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_task_manager",
+    srcs = [
+        "src/ray/raylet/scheduling/cluster_task_manager.cc",
+        "src/ray/raylet/scheduling/scheduler_stats.cc",
+    ],
+    hdrs = [
+        "src/ray/raylet/scheduling/cluster_task_manager.h",
+        "src/ray/raylet/scheduling/scheduler_stats.h",
+    ],
+    deps = [
+        ":cluster_resource_scheduler",
+        ":cluster_task_manager_interface",
+        ":local_task_manager_interface",
+        ":scheduler_resource_reporter",
+    ],
+)
+
+ray_cc_library(
+    name = "cluster_task_manager_interface",
+    hdrs = ["src/ray/raylet/scheduling/cluster_task_manager_interface.h"],
```

Comment:
please add dependencies for all headers
https://github.com/ray-project/ray/blob/1103bd114c3dcc4235d0fa8cbbd5b89f7d00dc99/src/ray/raylet/scheduling/cluster_task_manager_interface.h#L20-L21